src/wallet/wallet.cpp

```diff
@@ -180,13 +192,160 @@ int64_t CWallet::GetKeyCreationTime(const CBitcoinAddress& address)
 CBitcoinAddress CWallet::GenerateNewAutoMintKey()
 {
     CBitcoinAddress btcAddress;
-    CKeyID keyID = GenerateNewKey().GetID();
+    CKeyID keyID = GenerateNewKey(0).GetID();
     btcAddress.Set(keyID);
     CWalletDB(strWalletFile).WriteAutoConvertKey(btcAddress);
     SetAddressBook(keyID, "automint-address", "receive");
     setAutoConvertAddresses.emplace(btcAddress);
     return btcAddress;
 }
+void CWallet::DeriveNewChildKey(const CKeyMetadata& metadata, CKey& secretRet, uint32_t nAccountIndex, bool fInternal)
```

Comment:
Why this method is inside the wallet class and not in the hd chain class (or an intermediary HD chain family class)?